warehouse/query-core/src/main/java/datawave/next/async/ContextThreadFactory.java

```diff
@@ -0,0 +1,34 @@
+package datawave.next.async;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+import datawave.next.scanner.DocumentRangeScan;
+import datawave.next.scanner.QueryDataConsumer;
+
+/**
+ * A {@link ThreadFactory} that allows threads to be created with a default name, or context.
+ * <p>
+ * It is expected that the {@link QueryDataConsumer} or {@link DocumentRangeScan} will update their thread name with something more specific.
+ */
+public class ContextThreadFactory implements ThreadFactory {
+
+    private final String context;
+    private final ThreadFactory threadFactory;
+    private final ContextualUncaughtExceptionHandler uncaughtExceptionHandler;
+
+    public ContextThreadFactory(String context) {
+        this.context = context;
+        threadFactory = Executors.defaultThreadFactory();
+        uncaughtExceptionHandler = new ContextualUncaughtExceptionHandler();
```

Comment:
I think this has to be linked up with the existing uncaughtExceptionHandler associated with the query

Reply:
Created https://github.com/NationalSecurityAgency/datawave/issues/2910